### PR TITLE
Chat module

### PR DESCRIPTION
**Pull request by joedaouddotcom**
Create the chat module to display messages. Created the module, view and controller.

### CODE REVIEW

**Comment by RafGDev** [comment]
The changes in chatRoutes.js does not comply with the rest of the Api. Success messages should look like `res.status(status).json{"response": "success", "data": data_here}`.

**Comment by JGefroh** [comment]
The client-side JS library for socket.io comes with socket.io and is delivered with a special endpoint - it doesn't need to be included in the frontend as an additional JS file.

File: frontend/app/modules/chat/chat-service.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          return messagesArray; // Returns data in an array
+        })
+        };
+
+      service.sendMessage = function(data) {
+        return $http({
+          method: "POST",
+          url: "/api/sendChatMessage",
+          data: $.param({channelId: data.channelId, message: data.message}),
+          headers: {
```
Comment (JGefroh) [inline frontend/app/modules/chat/chat-service.js:35]: Headers should be moved into interceptors so they are centralized.

File: frontend/app/modules/chat/chat-service.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            Authorization: "Bearer " + Cookies.get("auth"),
+          }
+        })
+        .then(function(res) {
+          const messagesArray = [];
+
+          res.data.data.forEach(function(val) {
+            val.contents = JSON.parse(val.contents); //Parses the stringified string
+            messagesArray.push(val);
```
Comment (JGefroh) [inline frontend/app/modules/chat/chat-service.js:23]: Not entirely sure why this has to be done - normally you can access `response.data` as a complete json object.

**Comment by JGefroh** [comment]
After this PR, I recommend thinking of ways to clean up the backend structure - it's getting a bit unwieldy.

File: frontend/app/modules/chat/chat-service.js (hunk 2 above)
Comment (RafGDev) [reply frontend/app/modules/chat/chat-service.js:23]: I have sent the chat message as a JSON.stringified string. So e.g. '"hello \n raf"'. This way I could preserve formatting like new lines. This means that I also need to parse the contents of the string on the clientside again.

**Pull request merged**